components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/UserSessionManagementServiceImpl.java

```diff
@@ -57,17 +61,71 @@ public void terminateSessionsOfUser(String username, String userStoreDomain, Str
             UserSessionException {
 
         validate(username, userStoreDomain, tenantDomain);
-        List<String> sessionListOfUser = getSessionsOfUser(username, userStoreDomain, tenantDomain);
 
-        if (!sessionListOfUser.isEmpty()) {
-            if (log.isDebugEnabled()) {
-                log.debug("Terminating all the active sessions of user: " + username + " of userstore domain: " +
-                        userStoreDomain + " in tenant: " + tenantDomain);
+        String userId = resolveUserIdFromUsername(getTenantId(tenantDomain), userStoreDomain, username);
+        try {
+            terminateSessionsByUserId(userId);
+        } catch (SessionManagementException e) {
+            throw new UserSessionException("Error while terminating sessions of user.", e);
+        }
+    }
+
+    /**
+     * Retrieves the unique user id of the given username.
+     *
+     * @param tenantId          id of the tenant domain of the user
+     * @param userStoreDomain   userstore of the user
+     * @param username          username
+     * @return                  unique user id of the user
+     * @throws UserSessionException
+     */
+    private String resolveUserIdFromUsername(int tenantId, String userStoreDomain, String username) throws
+            UserSessionException {
+
+        try {
+            if (userStoreDomain == null) {
+                userStoreDomain = UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME;
+            }
+            UserStoreManager userStoreManager = getUserStoreManager(tenantId, userStoreDomain);
+            try {
+                if (userStoreManager instanceof AbstractUserStoreManager) {
+                    return ((AbstractUserStoreManager) userStoreManager).getUserIDFromUserName(username);
+                }
+                if (log.isDebugEnabled()) {
+                    log.debug("Provided user store manager for the user: " + username + ", is not an instance of the " +
```

Comment:
Would be better to have the userstoreDomain included as a debug log

Reply:
Fixed 8a7f2322bf7980e1b678bc34667f82699d5d1ce1